internal/pkg/gateway/commit/channelnotifier.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2021 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit
+
+import (
+	"sync"
+
+	"github.com/hyperledger/fabric/core/ledger"
+)
+
+type channelNotifier struct {
+	lock          sync.Mutex
+	commitChannel <-chan *ledger.CommitNotification
+	listeners     map[*channelListener]bool
+}
+
+func newChannelNotifier(commitChannel <-chan *ledger.CommitNotification) *channelNotifier {
+	notifier := &channelNotifier{
+		commitChannel: commitChannel,
+		listeners:     make(map[*channelListener]bool),
+	}
+	go notifier.run()
+	return notifier
+}
+
+func (channelNotifier *channelNotifier) run() {
```

Comment:
Why aren't we using idiomatic names? This stutters quite a bit and confuses the self-reference with the type name.

Reply:
Outdated